src/images.js

```diff
@@ -1,81 +1,81 @@
-  return http.request({
-    method: 'get',
-    url: '/images'
-  })
-}
+export default http => new Images(http);
 
-const getImage = (
-  http,
-  { id, returns, size, backgroundSize, choiceSize }
-) => {
-  let requestQuery = {
-    method: 'get',
-    url: `/images/${id}`
+class Images {
+  constructor(_http) {
+    this._http = _http;
   }
 
-  if (returns === 'json') {
-    requestQuery.headers = {
-      Accept: 'application/json'
-    }
+  add({ image, mediaType, fileName}  = {}) {
+    return this._http.request({
+      method: 'post',
+      url: `/images`,
+      data: {
+        image,
+        file_name: fileName,
+        media_type: mediaType
+      }
+    });
   }
 
-  if (size !== undefined) {
-    if (['default', 'thumbnail', 'mobile'].includes(size)) {
-      requestQuery['url'] += `/image/${size}`
-    } else {
-      throw new Error(`Image size doesn't exists`)
-    }
+  delete({ id }  = {}) {
+    return this._http.request({
+      method: 'delete',
+      url: `/images/${id}`
+    });
   }
 
-  if (backgroundSize !== undefined) {
-    if (['default', 'thumbnail', 'mobile', 'tablet'].includes(backgroundSize)) {
-      requestQuery['url'] += `/background/${backgroundSize}`
-    } else {
-      throw new Error(`Image background size doesn't exists`)
+  get({ id, returns, size, backgroundSize, choiceSize }  = {}) {
+    const requestQuery = {
+      method: 'get',
+      url: `/images/${id}`
+    };
+  
+    if (returns === 'json') {
+      requestQuery.headers = {
+        Accept: 'application/json'
+      };
     }
-  }
-
-  const choiceImageSizes = [
-    'default',
-    'thumbnail',
-    'supersize',
-    'supermobile',
-    'supersizefit',
-    'supermobilefit'
-  ]
-  if (choiceSize !== undefined) {
-    if (choiceImageSizes.includes(choiceSize)) {
-      requestQuery['url'] += `/choice/${choiceSize}`
-    } else {
-      throw new Error(`Image choice size doesn't exists`)
+  
+    if (size !== undefined) {
+      if (['default', 'thumbnail', 'mobile'].includes(size)) {
+        requestQuery.url += `/image/${size}`;
+      } else {
+        throw new Error(`Image size doesn't exists`);
+      }
     }
-  }
-
-  return http.request(requestQuery)
-}
+  
+    if (backgroundSize !== undefined) {
+      if (['default', 'thumbnail', 'mobile', 'tablet'].includes(backgroundSize)) {
+        requestQuery.url += `/background/${backgroundSize}`;
+      } else {
+        throw new Error(`Image background size doesn't exists`);
+      }
+    }
+  
+    const choiceImageSizes = [
+      'default',
+      'thumbnail',
+      'supersize',
+      'supermobile',
+      'supersizefit',
+      'supermobilefit'
+    ];
 
-const addImage = (http, { image, mediaType, fileName }) => {
-  return http.request({
-    method: 'post',
-    url: `/images`,
-    data: {
-      image,
-      file_name: fileName,
-      media_type: mediaType
+    if (choiceSize !== undefined) {
+      if (choiceImageSizes.includes(choiceSize)) {
```

Comment:
We could join this condition here as:

```js
if (typeof choiceSize !== 'undefined' && choiceImageSizes.includes(choiceSize)) { 
  // ... 
} else {
  throw Error()
}
```

Reply:
Then we do an elseif for `typeof choiceSize !== 'undefined' && !choiceImageSizes.includes(choiceSize)`?

Reply:
I would just do else since you want to throw the error anyways. Is there a situation where the user doesn't provide a valid image size that you don't want to throw?

Reply:
So the `get` method was designed to fetch an image by its id as well as fetch a specific image by a size. So ideally a user can call the method without giving it a size. Aka `[https://api.typeform.com/images/{image_id}
](https://developer.typeform.com/create/reference/retrieve-image/)` AND `[https://api.typeform.com/images/{image_id}/image/{size}](https://developer.typeform.com/create/reference/retrieve-image-by-size/)`.

So there's a real chance that they may not pass in a `choiceSize`, and if I did:

```
if (typeof choiceSize !== 'undefined' && choiceImageSizes.includes(choiceSize)) { 
  // ... 
} else {
  throw Error()
}
```

The first condition would fail, and an invalid error would be thrown.